src/backend/cpu/topk.cpp

```diff
@@ -60,13 +62,13 @@ void topk(Array<T>& vals, Array<unsigned>& idxs, const Array<T>& in,
                 partial_sort_copy(
                     idx_itr, idx_itr + in.strides()[1], kiptr, kiptr + k,
                     [ptr](const uint lhs, const uint rhs) -> bool {
-                        return ptr[lhs] < ptr[rhs];
+                        return compute_t<T>(ptr[lhs]) < compute_t<T>(ptr[rhs]);
```

Comment:
Probably some other backend header is including `types.hpp`. I don't see the header included in this source file but yet we still use `compute_t` alias in here. It may fail compilation in one or the other compiler, most  probably MSVC.